From 1c605a69e0797b79def4111d7855befe9608eda9 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Sun, 28 Aug 2011 20:43:26 -0700
Subject: [PATCH] build: fix library/api versioning information

VTE_API_VERSION is used alongwith a preceeding dash to generate the pc
(pkgconfig) files as well as the library names.  The other macros are no longer
used due to the hard requirement of GTK+ 3 due to the accessibility cleanups in
3bdb0639aa26ce1dcd745bab71a6bd5dddca6019.

Simply remove the extra macros.  -DVTE_SEAL_ENABLED is now always passed, so
simply inline the references to it.  Only the 3.0 API version for GTK+ is
supported, so inline that as well.
---
 Makefile.am           |    2 +-
 configure.in          |   39 +++++++++++++++++----------------------
 src/Makefile.am       |    2 +-
 vte-uninstalled.pc.in |    2 +-
 vte.pc.in             |    2 +-
 5 files changed, 21 insertions(+), 26 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 91c7109..5a24ede 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -19,7 +19,7 @@ SUBDIRS += glade
 endif
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = vte$(VTE_PC_VERSION).pc
+pkgconfig_DATA = vte-$(VTE_API_VERSION).pc
 
 DISTCHECK_CONFIGURE_FLAGS = \
 	--enable-gtk-doc \
diff --git a/configure.in b/configure.in
index e89106d..13effac 100644
--- a/configure.in
+++ b/configure.in
@@ -35,27 +35,22 @@ LT_VERSION_INFO="lt_triplet()"
 AC_SUBST([LT_VERSION_INFO])
 
 ################################################################################
+# VTE API/library versioning
+################################################################################
+
+VTE_API_VERSION=2.90
+AC_SUBST([VTE_API_VERSION])
+
+AC_SUBST([VTE_API_VERSION_U],[AS_TR_SH([$VTE_API_VERSION])])
+AC_SUBST([VTE_LIBRARY_SUFFIX_U],[AS_TR_SH([$VTE_API_VERSION])])
+
+################################################################################
 # GTK+
 ################################################################################
 
-VTE_API_MAJOR_VERSION=2
-VTE_API_MINOR_VERSION=90
-VTE_PC_VERSION=-$VTE_API_VERSION
-VTE_LIBRARY_SUFFIX=$VTE_API_VERSION
-VTE_SEAL_CFLAGS="-DVTE_SEAL_ENABLE"
-GTK_API_VERSION=3.0
 GTK_REQUIRED=3.1.9
 GLADE_API_VERSION=2.0
 
-AC_SUBST([VTE_API_VERSION])
-AC_SUBST([VTE_API_VERSION_U],[AS_TR_SH([$VTE_API_VERSION])])
-AC_SUBST([VTE_API_MAJOR_VERSION])
-AC_SUBST([VTE_API_MINOR_VERSION])
-AC_SUBST([VTE_PC_VERSION])
-AC_SUBST([VTE_SEAL_CFLAGS])
-AC_SUBST([GTK_API_VERSION])
-AC_SUBST([VTE_LIBRARY_SUFFIX_U],[AS_TR_SH([$VTE_LIBRARY_SUFFIX])])
-
 ################################################################################
 
 # Check for programs
@@ -297,7 +292,7 @@ GLIB_REQUIRED=2.26.0
 PANGO_REQUIRED=1.22.0
 AC_DEFINE(GDK_MULTIHEAD_SAFE,1,[Force use of GDK's multihead-safe APIs.])
 PKG_CHECK_MODULES(GLIB,[glib-2.0 >= $GLIB_REQUIRED gobject-2.0])
-PKG_CHECK_MODULES(GTK,[glib-2.0 >= $GLIB_REQUIRED gobject-2.0 gtk+-$GTK_API_VERSION >= $GTK_REQUIRED])
+PKG_CHECK_MODULES(GTK,[glib-2.0 >= $GLIB_REQUIRED gobject-2.0 gtk+-3.0 >= $GTK_REQUIRED])
 
 AC_PATH_PROG([GLIB_GENMARSHAL],[glib-genmarshal])
 AC_PATH_PROG([GLIB_MKENUMS],[glib-mkenums])
@@ -379,14 +374,14 @@ esac
 
 # We have a direct dependency on X11 on gdk-x11, see bug #613525
 AC_MSG_CHECKING([for GDK target])
-GDK_TARGET="$($PKG_CONFIG --variable target gdk-$GTK_API_VERSION)"
+GDK_TARGET="$($PKG_CONFIG --variable target gdk-3.0)"
 AC_MSG_RESULT([$GDK_TARGET])
 case "$GDK_TARGET" in
   x11) PLATFORM_PKGS="x11 cairo-xlib" ;;
   *) PLATFORM_PKGS="" ;;
 esac
 
-VTE_PKGS="glib-2.0 >= $GLIB_REQUIRED gobject-2.0 pango >= $PANGO_REQUIRED gtk+-$GTK_API_VERSION >= $GTK_REQUIRED gobject-2.0 gio-2.0 gio-unix-2.0 $PLATFORM_PKGS"
+VTE_PKGS="glib-2.0 >= $GLIB_REQUIRED gobject-2.0 pango >= $PANGO_REQUIRED gtk+-3.0 >= $GTK_REQUIRED gobject-2.0 gio-2.0 gio-unix-2.0 $PLATFORM_PKGS"
 PKG_CHECK_MODULES([VTE],[$VTE_PKGS])
 AC_SUBST([VTE_PKGS])
 
@@ -461,7 +456,7 @@ GOBJECT_INTROSPECTION_CHECK([0.9.0])
 GTK_DOC_CHECK([1.13],[--flavour no-tmpl])
 
 AC_SUBST([GLIB_PREFIX],[$($PKG_CONFIG --variable=prefix glib-2.0)])
-AC_SUBST([GTK_PREFIX],[$($PKG_CONFIG --variable=prefix gtk+-$GTK_API_VERSION)])
+AC_SUBST([GTK_PREFIX],[$($PKG_CONFIG --variable=prefix gtk+-3.0)])
 
 ################################################################################
 
@@ -485,8 +480,8 @@ doc/reference/version.xml
 glade/Makefile
 ])
 
-AC_CONFIG_FILES([vte${VTE_PC_VERSION}.pc:vte.pc.in],[VTE_PC_VERSION=${VTE_PC_VERSION}])
-AC_CONFIG_FILES([vte${VTE_PC_VERSION}-uninstalled.pc:vte-uninstalled.pc.in],[VTE_PC_VERSION=${VTE_PC_VERSION}])
+AC_CONFIG_FILES([vte-${VTE_API_VERSION}.pc:vte.pc.in])
+AC_CONFIG_FILES([vte-${VTE_API_VERSION}-uninstalled.pc:vte-uninstalled.pc.in])
 
 AC_OUTPUT
 
@@ -499,7 +494,7 @@ cat <<EOF | tee -a config.log
 ## Configuration. ##
 ## -------------- ##
 
-Configuration for libvte $VERSION for gtk+-$GTK_API_VERSION
+Configuration for libvte $VERSION for gtk+-3.0
 	Installing Glade catalogue: $enable_glade_catalogue
 	Debugging: $DEBUG
 	Disallow deprecated features: $enable_deprecation
diff --git a/src/Makefile.am b/src/Makefile.am
index 1971458..4fb4244 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -168,7 +168,7 @@ INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
 if HAVE_INTROSPECTION
 
 Vte-@VTE_API_VERSION@.gir: libvte@VTE_LIBRARY_SUFFIX_U@.la
-Vte_@VTE_API_VERSION_U@_gir_INCLUDES = Pango-1.0 Gdk-$(GTK_API_VERSION) Gtk-$(GTK_API_VERSION)
+Vte_@VTE_API_VERSION_U@_gir_INCLUDES = Pango-1.0 Gdk-3.0 Gtk-3.0
 Vte_@VTE_API_VERSION_U@_gir_CFLAGS = $(VTE_CFLAGS) -DVTE_COMPILATION -I$(top_srcdir)
 Vte_@VTE_API_VERSION_U@_gir_LIBS = libvte@VTE_LIBRARY_SUFFIX_U@.la
 Vte_@VTE_API_VERSION_U@_gir_FILES = \
diff --git a/vte-uninstalled.pc.in b/vte-uninstalled.pc.in
index 36ee26b..ed4a5c8 100644
--- a/vte-uninstalled.pc.in
+++ b/vte-uninstalled.pc.in
@@ -10,4 +10,4 @@ Description: Vte terminal widget.
 Version: @VERSION@
 Requires: @VTE_PKGS@
 Libs: ${pc_top_builddir}/${pcfiledir}/src/libvte@VTE_LIBRARY_SUFFIX_U@.la
-Cflags: -I${pc_top_builddir}/${pcfiledir}/src @VTE_SEAL_CFLAGS@
+Cflags: -I${pc_top_builddir}/${pcfiledir}/src -DVTE_SEAL_ENABLE
diff --git a/vte.pc.in b/vte.pc.in
index 672171a..da45192 100644
--- a/vte.pc.in
+++ b/vte.pc.in
@@ -10,4 +10,4 @@ Description: Vte terminal widget.
 Version: @VERSION@
 Requires: @VTE_PKGS@
 Libs: -L${libdir} -lvte@VTE_LIBRARY_SUFFIX_U@
-Cflags: -I${includedir}/vte-@VTE_API_VERSION@ @VTE_SEAL_CFLAGS@
+Cflags: -I${includedir}/vte-@VTE_API_VERSION@ -DVTE_SEAL_ENABLE
-- 
1.7.6.1

